api/src/org/apache/cloudstack/api/command/user/vm/DeployVMCmd.java

```diff
@@ -443,6 +449,37 @@ public String getKeyboard() {
         return dhcpOptionsMap;
     }
 
+    public Map<Long, DiskOffering> getDataDiskTemplateToDiskOfferingMap() {
+        if (diskOfferingId != null && dataDiskTemplateToDiskOfferingList != null) {
+            throw new InvalidParameterValueException("diskofferingid paramter can't be specified along with datadisktemplatetodiskofferinglist parameter");
+        }
+        HashMap<Long, DiskOffering> dataDiskTemplateToDiskOfferingMap = new HashMap<Long, DiskOffering>();
+        if (dataDiskTemplateToDiskOfferingList != null && !dataDiskTemplateToDiskOfferingList.isEmpty()) {
+            Collection dataDiskTemplatesCollection = dataDiskTemplateToDiskOfferingList.values();
+            Iterator iter = dataDiskTemplatesCollection.iterator();
+            while (iter.hasNext()) {
```

Comment:
why not a for each?
```
for( HashMap<String, String> dataDiskTemplates:  dataDiskTemplateToDiskOfferingList.values()){
...
}
```

The use of Iterator manually does not seem to be necessary.